File: core/src/main/java/com/views/animation/flyschool/AppCompatShapeView.java
package com.views.animation.flyschool;

import android.content.Context;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatImageView;

import com.R;

public class AppCompatShapeView extends AppCompatImageView implements ShapeSetter {
    public AppCompatShapeView(Context context) {
        super(context);
        init();
    }

    public AppCompatShapeView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public AppCompatShapeView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        setImageResource(R.drawable.l_facebook_button_blue);
    }

    @Override
    public void setShape(int drawable) {
        setImageResource(drawable);
        Utils.setHeart(this);
    }

    @Override
    public void setShape(ImgObject imgObject, int drawableRes) {
        //setImageResource(R.mipmap.ic_launcher);
    }
}

File: core/src/main/java/com/loitp/anim/flySchool/AppCompatShapeView.kt
package com.loitp.anim.flySchool

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView
import com.loitp.R
import com.loitp.anim.flySchool.Utils.setHeart

class AppCompatShapeView : AppCompatImageView, ShapeSetter {
    constructor(context: Context) : super(context) {
        init()
    }

    constructor(
        context: Context,
        attrs: AttributeSet?
    ) : super(context, attrs) {
        init()
    }

    constructor(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int
    ) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        init()
    }

    private fun init() {
        setImageResource(R.drawable.l_facebook_button_blue)
    }

    override fun setShape(drawable: Int) {
        setImageResource(drawable)
        setHeart(this)
    }

    override fun setShape(
        imgObject: ImgObject,
        drawableRes: Int
    ) {
        // setImageResource(R.mipmap.ic_launcher);
    }
}
